src/main/java/frc/trigon/robot/subsystems/turret/kablamaturret/KablamaTurretIO.java

```diff
@@ -0,0 +1,38 @@
+package frc.trigon.robot.subsystems.turret.kablamaturret;
+
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import frc.trigon.robot.subsystems.turret.TurretIO;
+import frc.trigon.robot.subsystems.turret.TurretInputsAutoLogged;
+
+public class KablamaTurretIO extends TurretIO {
+    private final TalonFX motor = KablamaTurretConstants.MOTOR;
+    private final MotionMagicVoltage positionVoltageRequest = new MotionMagicVoltage(0).withEnableFOC(KablamaTurretConstants.FOC_ENABLED);
+
+    @Override
+    protected void updateInputs(TurretInputsAutoLogged inputs) {
+        inputs.motorPositionDegrees = getPosition().getDegrees();
+        inputs.motorVelocityDegreesPerSecond = getVelocityDegreesPerSecond();
+        inputs.motorVoltage = motor.getMotorVoltage().refresh().getValue();
+    }
```

Comment:
update the profiled target position using the `getClosedLoopReference` status signal.